biosnoop.py and biotop.py preprocessor fails on kernel 4.10 due to undefined REQ_OP_SHIFT

REQ_OP_SHIFT was killed in https://github.com/torvalds/linux/commit/ef295ecf090d3e86e5b742fc6ab34f1122a43773 which is in 4.10. 

```
linux $ git tag --contains ef295ecf090d3e86e5b742fc6ab34f1122a43773
v4.10-rc1
v4.10-rc2
```

The result is:
```
[root@localhost tools]# python biotop.py
In file included from /virtual/main.c:3:
In file included from /lib/modules/4.10.0-0.rc2.git3.2.fc26.x86_64/build/include/linux/blkdev.h:9:
In file included from /lib/modules/4.10.0-0.rc2.git3.2.fc26.x86_64/build/include/linux/genhd.h:64:
In file included from /lib/modules/4.10.0-0.rc2.git3.2.fc26.x86_64/build/include/linux/device.h:24:
In file included from /lib/modules/4.10.0-0.rc2.git3.2.fc26.x86_64/build/include/linux/pinctrl/devinfo.h:21:
In file included from /lib/modules/4.10.0-0.rc2.git3.2.fc26.x86_64/build/include/linux/pinctrl/consumer.h:17:
In file included from /lib/modules/4.10.0-0.rc2.git3.2.fc26.x86_64/build/include/linux/seq_file.h:10:
/lib/modules/4.10.0-0.rc2.git3.2.fc26.x86_64/build/include/linux/fs.h:2648:9: warning: comparison of unsigned enum expression < 0 is always false [-Wtautological-compare]
        if (id < 0 || id >= READING_MAX_ID)
            ~~ ^ ~
/virtual/main.c:84:41: error: use of undeclared identifier 'REQ_OP_SHIFT'
    info.rwflag = !!((req->cmd_flags >> REQ_OP_SHIFT) == REQ_OP_WRITE);
                                        ^
1 warning and 1 error generated.
Traceback (most recent call last):
  File "biotop.py", line 164, in <module>
    """, debug=0)
  File "/usr/lib/python2.7/site-packages/bcc/__init__.py", line 240, in __init__
    raise Exception("Failed to compile BPF module %s" % src_file)
Exception: Failed to compile BPF module
```
```
[root@localhost tools]# python biosnoop.py
In file included from /virtual/main.c:3:
In file included from /lib/modules/4.10.0-0.rc2.git3.2.fc26.x86_64/build/include/linux/blkdev.h:9:
In file included from /lib/modules/4.10.0-0.rc2.git3.2.fc26.x86_64/build/include/linux/genhd.h:64:
In file included from /lib/modules/4.10.0-0.rc2.git3.2.fc26.x86_64/build/include/linux/device.h:24:
In file included from /lib/modules/4.10.0-0.rc2.git3.2.fc26.x86_64/build/include/linux/pinctrl/devinfo.h:21:
In file included from /lib/modules/4.10.0-0.rc2.git3.2.fc26.x86_64/build/include/linux/pinctrl/consumer.h:17:
In file included from /lib/modules/4.10.0-0.rc2.git3.2.fc26.x86_64/build/include/linux/seq_file.h:10:
/lib/modules/4.10.0-0.rc2.git3.2.fc26.x86_64/build/include/linux/fs.h:2648:9: warning: comparison of unsigned enum expression < 0 is always false [-Wtautological-compare]
        if (id < 0 || id >= READING_MAX_ID)
            ~~ ^ ~
/virtual/main.c:90:24: error: use of undeclared identifier 'REQ_OP_SHIFT'
if ((req->cmd_flags >> REQ_OP_SHIFT) == REQ_OP_WRITE) {
                       ^
1 warning and 1 error generated.
Traceback (most recent call last):
  File "biosnoop.py", line 124, in <module>
    """, debug=0)
  File "/usr/lib/python2.7/site-packages/bcc/__init__.py", line 240, in __init__
    raise Exception("Failed to compile BPF module %s" % src_file)
Exception: Failed to compile BPF module

```

I'll send a PR.

--- a/tools/biosnoop.py
+++ b/tools/biosnoop.py
@@ -106,14 +106,12 @@ int trace_req_completion(struct pt_regs *ctx, struct request *req)
  * test, and maintenance burden.
  */
 #ifdef REQ_WRITE
-if (req->cmd_flags & REQ_WRITE) {
+    data.rwflag = !!(req->cmd_flags & REQ_WRITE);
+#elif defined(REQ_OP_SHIFT)
+    data.rwflag = !!((req->cmd_flags >> REQ_OP_SHIFT) == REQ_OP_WRITE);
 #else
-if ((req->cmd_flags >> REQ_OP_SHIFT) == REQ_OP_WRITE) {
+    data.rwflag = !!((req->cmd_flags & REQ_OP_MASK) == REQ_OP_WRITE);
 #endif
-        data.rwflag = 1;
-    } else {
-        data.rwflag = 0;
-    }
 
     events.perf_submit(ctx, &data, sizeof(data));
     start.delete(&req);

--- a/tools/biotop.py
+++ b/tools/biotop.py
@@ -137,8 +137,10 @@ int trace_req_completion(struct pt_regs *ctx, struct request *req)
  */
 #ifdef REQ_WRITE
     info.rwflag = !!(req->cmd_flags & REQ_WRITE);
-#else
+#elif defined(REQ_OP_SHIFT)
     info.rwflag = !!((req->cmd_flags >> REQ_OP_SHIFT) == REQ_OP_WRITE);
+#else
+    info.rwflag = !!((req->cmd_flags & REQ_OP_MASK) == REQ_OP_WRITE);
 #endif
 
     whop = whobyreq.lookup(&req);

